zerver/views/registration.py

```diff
@@ -170,10 +176,17 @@ def accounts_register(
     if realm_creation:
         role = UserProfile.ROLE_REALM_OWNER
 
-    try:
-        validators.validate_email(email)
-    except ValidationError:
-        return render(request, "zerver/invalid_email.html", context={"invalid_email": True})
+    if realm_creation and is_demo_organization:
+        # Demo organization owners are not required to provide an email
+        # upon demo realm creation, so we can skip email validation
+        # for demo org owners and the email can be blank till the owner
+        # sets it when converting the demo organization.
+        email = ""
+    else:
+        try:
+            validators.validate_email(email)
+        except ValidationError:
+            return render(request, "zerver/invalid_email.html", context={"invalid_email": True})
```

Comment:
I think we don't want this hunk yet, since our plan is to avoid making it possible to create demo organizations except via the development environment button, and this would do so.